src/core/namespace.c

```diff
@@ -168,16 +167,21 @@ static int append_mounts(BindMount **p, char **strv, MountMode mode) {
 
         STRV_FOREACH(i, strv) {
                 bool ignore = false;
+                char *path;
 
                 if (IN_SET(mode, INACCESSIBLE, READONLY, READWRITE) && startswith(*i, "-")) {
                         (*i)++;
                         ignore = true;
                 }
 
-                if (!path_is_absolute(*i))
+                path = strdup(*i);
+                if (!path)
+                        return -ENOMEM;
+
+                if (!path_is_absolute(path))
```

Comment:
Oh, I see. So
https://github.com/systemd/systemd/pull/4594/files#diff-f315b72505d5f0a92ec2f4d068d4f916L746
```c
                 r = append_mounts(&m, read_write_paths, READWRITE);
                  if (r < 0)
                          return r;
```
should be
```c
                 r = append_mounts(&m, read_write_paths, READWRITE);
                  if (r < 0)
                          goto finish;
```
Right?